examples/sub-class.py

```diff
@@ -29,36 +29,30 @@
         sys.path.insert(0, cmd_subfolder)
     import paho.mqtt.client as mqtt
 
-class MyMQTTClass:
-    def __init__(self, clientid=None):
-        self._mqttc = mqtt.Client(clientid)
-        self._mqttc.on_message = self.mqtt_on_message
-        self._mqttc.on_connect = self.mqtt_on_connect
-        self._mqttc.on_publish = self.mqtt_on_publish
-        self._mqttc.on_subscribe = self.mqtt_on_subscribe
+class MyMQTTClass(mqtt.Client):
 
-    def mqtt_on_connect(self, mqttc, obj, flags, rc):
+    def on_connect(self, mqttc, obj, flags, rc):
```

Comment:
So, I've been thinking a lot about this, and, I now think that the current implementation (with or without this pull request) is the "right" behaviour, since, in this scenario where an instance method is used as the callback, the first argument is the "application" object and the second is the "MQTT client" object.

In particular, IMO, implementing the callbacks by inheriting from the `Client` class is a "bad" design choice and a "better" design is to use composition, i.e. your application should "use" a MQTT client object rather than "be" a MQTT client, since this gives better encapsulation and SoC. You could even use DI to inject the MQTT client. This is how the "subclassing" example currently works, except that `MyMqttClass` should be `MyMqttApp` or similar, and I would suggest to maintain that example as a "composition" example.

I'm sorry to interfere; I'm just an interested member of the community.